aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -554,15 +816,165 @@ module aptos_framework::delegation_pool {
+    inline fun borrow_mut_used_voting_power(governance_records: &mut GovernanceRecords, voter: address, proposal_id: u64): &mut u64 {
+        let votes = &mut governance_records.votes;
+        let key = VotingRecordKey {
+            proposal_id,
+            voter,
+        };
+        smart_table::borrow_mut_with_default(votes, key, 0)
+    }
+
+    /// Get the latest VoteDelegation of a delegator.
+    /// The 2nd bool return value is true when VoteDelegation has already been created before calling this function.
+    /// The 3rd bool return value is true when the latest VoteDelegation is different from the existing VoteDelegation.
+    fun get_latest_delegator_vote_delegation(pool: &DelegationPool, governance_records: &GovernanceRecords, delegator: address): (VoteDelegation, bool, bool) {
+        let pool_address = get_pool_address(pool);
+        let locked_until_secs = stake::get_lockup_secs(pool_address);
+        let vote_delegation = &governance_records.vote_delegation;
+        // By default, a delegator's delegated voter is itself.
+        if (!smart_table::contains(vote_delegation, delegator)) {
+            return (VoteDelegation {
+                voter: delegator,
+                last_locked_until_secs: locked_until_secs,
+                pending_voter: delegator,
+            }, false, false)
+        };
+        let vote_delegation = *smart_table::borrow(vote_delegation, delegator);
+        let needs_update = false;
+        if (vote_delegation.last_locked_until_secs < locked_until_secs &&
+            vote_delegation.voter != vote_delegation.pending_voter) {
+            vote_delegation.voter = vote_delegation.pending_voter;
+            needs_update = true;
+        };
+        (vote_delegation, true, needs_update)
+    }
+
+    /// Update VoteDelegation of a delegator to up-to-date then borrow_mut it.
+    fun update_and_borrow_mut_delegator_vote_delegation(
+        pool: &DelegationPool,
+        governance_records :&mut GovernanceRecords,
+        delegator: address
+    ): &mut VoteDelegation {
+        let (latest_vote_delegation, is_existed, needs_update) =
+            get_latest_delegator_vote_delegation(pool, governance_records, delegator);
+        let vote_delegation_table = &mut governance_records.vote_delegation;
+        if (!is_existed) {
+            smart_table::add(vote_delegation_table, delegator, latest_vote_delegation);
+        };
+        let vote_delegation = smart_table::borrow_mut(vote_delegation_table, delegator);
+        if (needs_update) {
+            *vote_delegation = latest_vote_delegation;
+        };
+        vote_delegation
+    }
+
+    /// Get the latest VoteDelegation of a voter.
+    /// The 2nd bool return value is true when VoteDelegation has already been created before calling this function.
+    /// The 3rd bool return value is true when the latest VoteDelegation is different from the existing VoteDelegation.
+    fun get_latest_vote_delegation(
```

Comment:
This looks so similar to get_latest_delegator_vote_delegation that I had to check twice that it's not the same function. Can we merge the 2 somehow?